plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -6829,6 +6835,35 @@ public void testDropCorruptedTableWithHiveRedirection()
         assertFalse(fileSystem.listFiles(tableLocation).hasNext(), "Table location should not exist");
     }
 
+    @Test(timeOut = 10_000)
+    public void testNoRetryWhenMetadataFileInvalid()
+            throws Exception
+    {
+        String tableName = "test_no_retry_when_metadata_file_invalid_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT 1 id", 1);
+
+        String tableLocation = getTableLocation(tableName);
+        String metadataFileLocation = getLatestMetadataLocation(fileSystem, tableLocation);
+
+        ObjectMapper mapper = JsonUtil.mapper();
+        JsonNode jsonNode = mapper.readValue(fileSystem.newInputFile(Location.of(metadataFileLocation)).newStream(), JsonNode.class);
+        ArrayNode fieldsNode = (ArrayNode) jsonNode.get("schemas").get(0).get("fields");
+        ObjectNode newFieldNode = fieldsNode.get(0).deepCopy();
+        // Add duplicate field to produce validation error while reading the metadata file
+        fieldsNode.add(newFieldNode);
+
+        String modifiedJson = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(jsonNode);
+        try (OutputStream outputStream = fileSystem.newOutputFile(Location.of(metadataFileLocation)).createOrOverwrite()) {
+            // Corrupt metadata file by overwriting the invalid metadata content
+            outputStream.write(modifiedJson.getBytes(UTF_8));
+        }
+        assertThatThrownBy(() -> query("SELECT * FROM " + tableName))
+                .hasMessage("Invalid metadata file for table tpch.%s".formatted(tableName));
+
+        assertUpdate("DROP TABLE " + tableName);
```

Comment:
https://github.com/trinodb/trino/blob/2dc0342ce81909924ad8a772d1cfe0af5e603ad9/plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java#L533-L536

Do you happen to remember why we've added the test for the situation on the smoke test and not on BaseIcebergConnectorTest ?
I'm not necessarily requesting the checks to be done here as well.

Reply:
The tests are also added in `BaseIcebergConnectorTest`.

https://github.com/trinodb/trino/blob/2dc0342ce81909924ad8a772d1cfe0af5e603ad9/plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java#L6634

We wanted to verify the corrupted table against all catalogs and smoke test runs against all catalogs so added smoke test.

Reply:
One more reason to test in the smoke test is to verify against the different file systems.